viscy/representation/lca.py

```
@@ -197,8 +237,33 @@ def train_and_test_linear_classifier(
         raise ValueError("Labels must have 1 dimension.")
     embeddings = torch.from_numpy(embeddings)
     data = LinearProbingDataModule(
-        embeddings, torch.from_numpy(labels), split_ratio, batch_size
+        embeddings,
+        torch.from_numpy(labels),
+        split_ratio,
+        batch_size,
+        use_smote=use_smote,
     )
     model = LinearClassifier(embeddings.shape[1], num_classes, lr)
     trainer.fit(model, data)
     trainer.test(model, data)
+
+    test_preds = model.test_predictions_cache
+
+    if save_predictions and csv_path is not None and merged_df is not None:
```

Silent error if not all three are true.